### ISSUE TITLE
Überschreiben bei Dateien

### ISSUE THREAD

**Issue by jenda77** [contao/core#8187]
Ich fände toll, wenn man die Funktion Überschreiben beim mehrfachen Bearbeiten auch bei Dateien hätte.

**Comment by asaage** [comment]
markiere alle, überschreibe Name = "xyz" -> aus 100 Dateien mach eine.... ?
gefährlich...
Aber für die Metaiformationen fänd ich das vielleicht auch gar nicht so verkehrt.

**Comment by Toflar** [comment]
Und welches Bild hat dieselben Meta-Informationen wie das andere? Vorausgesetzt man schreibt was sinnvolles und nicht "Bild mit Farben"? :D

**Comment by asaage** [comment]
auch wieder wahr...
Aber ich habe zb die MetaInformationen in einem Projekt um die Felder "Urheber" und "Datum" erweitert.
Weil das aber serialisiert gespeichert wird bekäme man trotzdem immer das komplette Widget - was ein überschreiben einzelner Meta-Felder wohl unmöglich macht.

**Comment by jenda77** [comment]
Natürlich würde das vor allem oder sogar nur die Megainformationen betreffen. Bei Dateibenennung wäre es natürlich klasse, wenn man bestimmen könnte, dass diese nummeriert werden könnten, wenn sie ähnlich heißen sollen (selbstverständlich geht es wesentlich leichter am Rechner, gerade am Mac mit Better Renamer wunderbar), aber warum sollte man das auch nicht per PHP wagen?

**Comment by Toflar** [comment]
Man müsste bei mehrere bearbeiten dann Sprache und Feld wählen, was ein bisschen dem Standard-Contao-Flow widerspricht...

**Comment by jenda77** [comment]
Ich kann mir vorstellen, dass man das davon abhängig machen könnte, oder?

**Comment by Toflar** [comment]
Können tut man alles, man muss nur entscheiden ob es sinnvoll ist und jemand muss es tun. Ich hatte bisher nie den Anwendungsfall.

**Comment by Samson1964** [comment]
Ein auch für mich wünschenswertes Feature. Oft habe ich ganze Bildergalerien und schreibe dann -zig Mal in der Bildunterschrift denselben Fotografen im Format "[Name]" rein, was ich im Template mit Regex konvertiere. 
P.S. Die Idee mit zusätzlichen Metafeldern hatte ich noch gar nicht.

**Comment by Toflar** [comment]
Das Standardvorgehen von Contao bedeutet halt, dass der komplette Wert überschrieben wird. Das würde im Falle des Meta-Wizards bedeuten, dass halt alle Felder und alle Sprachen überschrieben würden. Das wäre relativ einfach zu erreichen, denke ich. Aber sobald man irgendwie sagen muss "nur Titel" und "nur Deutsch", wird's tricky weil's halt einfach nicht Contao ist.

**Comment by jenda77** [comment]
Und wie wäre es, wenn man die Liste der bearbeiteten Felder nach Sprache darstellt (verschachtelt). Diese würden einfach von ihr abhängen und somit auch für den Nutzer als Gruppe dargestellt. Man müsste, so stelle ich mir das wenigstens vor, eine Zuweisung zu den Sprachen herstellen, damit die Metafelder von der jeweiligen Sprache abhängen.

**Comment by Toflar** [comment]
Wie gesagt, es ist nicht der Contao-Standard. Mehrere bearbeiten zeigt dir eine Liste von Feldern an die dann komplett überschrieben werden. Du würdest also "Metadaten" auswählen, auf "weiter" klicken und würdest dann den Metawizard erhalten. Wenn du speicherst würden die Daten des Metawizards genau so wie du sie geändert hast in alle ausgewählten Datei-Datensätzen geschrieben. Das ist der Contao-Standard. Was aber in dem speziellen Fall sinnvoll wäre, wäre eben eine Auswahl von "Metawizard" und danach ein oder mehrere Felder dieses Metawizards und noch die Sprache. Das macht aber nur für den Metawizard Sinn, nicht aber für bspw. den Dateinamen, weil der ist ja nicht sprachabhängig und besteht auch nur aus einem Wert. Auch nicht für andere Felder, die sich ja beliebig erweitern lassen. Der Fall für den Metawizard kommt sonst nirgendwo in Contao vor und man müsste explizit für diesen Fall eine Lösung bauen. Selbstverständlich könnte man das tun. Ich werd mich allerdings dem Thema bestimmt nicht annehmen, weil für mich Aufwand/Nutzen in absolut keinem Verhältnis stehen und ich aus dem Stegreif 400 wichtigere Dinge (für mich persönlich) nennen könnte :-)

**Comment by jenda77** [comment]
Klar, vielleicht kann man gleich das ganze Sprachkonzept überdenken, weil das Contao helfen würde, noch stärker international eingesetzt zu werden. 

Deine Liste würde ich gerne sehen. ;-) Sicherlich findet man da auch etwas, was die anderen Nutzer nützlich finden.
